### PR TITLE
Remove command

### PR DESCRIPTION
**Pull request by Kento-Bento**
In order for remove command to work, first ensure your database is remade by using the `recreate_tables` command, because some changes have been made to the models.py (added cascade ondelete)

When using Remove, recommended to first check out the database by: `falcon_multiqc remove --overview`

To remove a particular cohort (or multiple) and all assoicated rows do:
`falcon_multiqc remove -c MGRB`

To remove some batches - and all their associated rows, do:
`falcon_multiqc remove -b Cohort2  19_Aug20_CAGRF20021238-2 -b Cohort2 9_Oct20_CAGRF20021238-7`

NOTE: If you want to keep working with the database after testing out this command, it's probably easiest to just run recreate_tables command + bulk save everything back 

## PR Description
insert description here

## PR checklist
 - [ ] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Ensure the CI test suite passes (coming soon...).
 - [ ] Make sure your code lints (coming soon...).
 - [ ] `CHANGELOG.md` is updated (if >=v1.0.0)
 - [ ] `README.md` is updated

### CODE REVIEW

File: database/models.py [hunk 1]
```diff
@@ -18,9 +18,9 @@ class Sample(Base):
 
     id = Column(Integer, primary_key=True, nullable=False)
 
-    patient_id = Column(Integer, ForeignKey("patient.id"), nullable=True)
-    batch_id = Column(Integer, ForeignKey("batch.id"), nullable=False)
-    cohort_id = Column(String, ForeignKey("cohort.id"), nullable=False)
+    patient_id = Column(Integer, ForeignKey("patient.id", ondelete="CASCADE"), nullable=True)
+    batch_id = Column(Integer, ForeignKey("batch.id", ondelete="CASCADE"), nullable=False)
+    cohort_id = Column(String, ForeignKey("cohort.id", ondelete="CASCADE"), nullable=False)
```
Comment (lwratten) [inline database/models.py:23]: what does the `ondelete="CASCADE"` do?

File: falcon_multiqc/commands/remove.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+# Command for removing entries from database.
+# --overview to see all the cohorts with there respective batches and number of samples
+# --cohort option requires one argument 'cohortID' - removes all entries associated with that cohort throughout db
+# --batch option requires two arguments 'cohortID and Batch' - removes all entries associated with that batch throughout db
+# Note: although you can use multiple --batch or --cohort options in one command, if both are used together, only --cohort will commit
+
+@click.option("-c", "--cohort", multiple=True, required=False, help="Which cohort to remove from the database. E.g. <MGRB>")
+@click.option ("-b", "--batch", multiple=True, type=(str, str), required=False, help="Which batch you want to remove from the database. E.g. <MGRB BAB>")
+@click.option("-ov", "--overview", is_flag=True, required=False, help="Prints an overview of the number of samples in each batch/cohort.")
```
Comment (lwratten) [inline falcon_multiqc/commands/remove.py:15]: Have you tested this out along with the other commands? Also a really small and unnecessary thing but personally not a big fan of `-ov` but `-o` is confusing and I can't think of anything else 😝

**Review by lwratten — APPROVED** [review]
Overall looks good to me! I'm not familiar enough with the db side of things to notice anything that might be an issue, trust u and we can test it out once its been merged in 😃  I'll approve but might be good to get @Celinet21 to have a peek too if she has a chance

File: database/models.py (hunk 1 above)
Comment (Kento-Bento) [reply database/models.py:23]: It means if the parent table (the table with the corresponding primary key to those foreignkeys) was deleted, then these children tables will also be deleted (cascade) - and in sqlalchemy, the parent table is considered where the relationship was defined. It's also possible to delete children and cascade that up to parents, which are the changes to the Cohort/Batch tables

File: falcon_multiqc/commands/remove.py (hunk 2 above)
Comment (Kento-Bento) [reply falcon_multiqc/commands/remove.py:15]: lol yeah I didn't like it too, I'll just scrap the short hand haha

File: database/models.py (hunk 1 above)
Comment (lwratten) [reply database/models.py:23]: oh yep okay i get you!

File: falcon_multiqc/commands/remove.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import click
+from database.crud import session_scope
+from sqlalchemy.orm import Query
+from .query import print_overview
+from database.models import Base, Batch, Cohort
+
+# Command for removing entries from database.
```
Comment (Celinet21) [inline falcon_multiqc/commands/remove.py:7]: nit: sorry can we make this comment block a multiline comment like with """

**Review by Celinet21 — APPROVED** [review]
Cool stuff, just posted one nit comment on the style of comment.

**Pull request merged**